keychain/btcwallet.go

```diff
@@ -477,7 +477,9 @@ func (b *BtcWalletKeyRing) SignMessageSchnorr(keyLoc KeyLocator,
 	}
 
 	var digest []byte
-	if doubleHash {
+	if noHash {
+                digest = msg
```

Comment:
This is really quite dangerous, especially if the message here is arbitrary input provided by the user. If the message happens to be a transaction hash or a p2p message, one could extract funds from the node or create fake gossip traffic. I'm honestly not sure if we should even allow this at all.